lib/fibonacci.rb

```diff
@@ -8,5 +8,20 @@
 # Time complexity: ?
 # Space complexity: ?
 def fibonacci(n)
-  raise NotImplementedError
+  if n.nil? || n < 0
+    raise ArgumentError
+  end
+  if n == 0
+    return 0
+  else
+    p1 = 0
```

Comment:
[nit] Why `p`? (Not that you should change it, it's just not obvious to me.)

Reply:
You have a good point. Do you think naming it p3 would make it easy to understand, since it is the number subsequent to p2 ?

Reply:
I was actually curious why you were using `p1` and `p2` and `p3`. What's the "p" for?